### PR TITLE
Fix auth & wishlist

### PR DESCRIPTION
**Pull request by cond0r**
- fix auth & register after a regression was introduced by the Dynamic API PR, unable to test since auth was disabled on preview deployments.
- show the actual validation responses thrown by BigCommerce
- fix wishlist API endpoints/operations schema & normalization after Type updates PR
- fix customer schema validation

### CODE REVIEW

File: packages/bigcommerce/src/api/endpoints/signup/signup.ts [hunk 1]
```diff
@@ -22,28 +21,27 @@ const signup: SignupEndpoint['handlers']['signup'] = async ({
...
+      const message = Object.values(error.data.errors).join('<br />')
+      if (message) {
+        throw new CommerceAPIError(message, {
           status: 400,
-          code: 'duplicated_email',
+          code: 'invalid_request',
         })
       }
-    } else {
-      throw error
```
Comment (lfades) [inline packages/bigcommerce/src/api/endpoints/signup/signup.ts:36]: I understand improving the error handling here, but why was the duplicated email check removed?

File: packages/bigcommerce/src/api/operations/get-customer-wishlist.ts [hunk 2]
```diff
@@ -41,18 +39,22 @@ export default function getCustomerWishlistOperation({
...
 
     const wishlist = data[0]
 
     if (includeProducts && wishlist?.items?.length) {
-      const ids = wishlist.items
-        ?.map((item) => (item?.productId ? String(item?.productId) : null))
-        .filter((id): id is string => !!id)
+      const ids = []
+
+      for (let i = 0; i < wishlist.items.length; i++) {
```
Comment (lfades) [inline packages/bigcommerce/src/api/operations/get-customer-wishlist.ts:51]: This is better as a `for of`

File: packages/bigcommerce/src/api/operations/login.ts [hunk 3]
```diff
@@ -63,19 +63,13 @@ export default function loginOperation({
         cookie = cookie.replace(/; SameSite=none/gi, '; SameSite=lax')
       }
 
-      const prevCookie = response.headers.get('Set-Cookie')
-      const newCookie = concatHeader(prevCookie, cookie)
```
Comment (lfades) [inline packages/bigcommerce/src/api/operations/login.ts:67]: The import for `concatHeader` was not removed, and that util might not be needed anymore if it's no longer used here, also, why is this part changing?

File: packages/bigcommerce/src/api/utils/fetch-graphql-api.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ const fetchGraphqlApi: (getConfig: () => BigcommerceConfig) => GraphQLFetcher =
   async (
     query: string,
     { variables, preview } = {},
-    options: { headers?: HeadersInit } = {}
+    headers?: HeadersInit
```
Comment (lfades) [inline packages/bigcommerce/src/api/utils/fetch-graphql-api.ts:10]: I don't see this change as a good idea, having an options object that includes headers is far better IMO.

File: packages/commerce/src/api/utils/index.ts [hunk 5]
```diff
@@ -44,7 +44,9 @@ export const transformRequest = (req: NextApiRequest, path: string) => {
     body = JSON.stringify(req.body)
   }
 
-  return new NextRequest(`https://${req.headers.host}/api/commerce/${path}`, {
+  const url = new URL(req.url || '/', `https://${req.headers.host}`)
+
+  return new NextRequest(url, {
```
Comment (lfades) [inline packages/commerce/src/api/utils/index.ts:49]: The `path` parameter of this function is no longer used after this change, why was this changed? not mentioned in the description 🤔

File: packages/commerce/src/schemas/customer.ts [hunk 6]
```diff
@@ -5,14 +5,16 @@ export const getCustomerAddressBodySchema = z.object({
...
 export const customerSchema = z.object({
-  id: z.string(),
-  firstName: z.string(),
-  lastName: z.string(),
-  email: z.string().optional(),
-  phone: z.string().optional(),
-  company: z.string().optional(),
-  notes: z.string().optional(),
-  acceptsMarketing: z.boolean().optional(),
+  customer: z.object({
```
Comment (lfades) [inline packages/commerce/src/schemas/customer.ts:8]: Why is it needed to add a nested `customer` object in the customer schema?

**Review by lfades — CHANGES_REQUESTED** [review]
Added multiple comments.

File: packages/bigcommerce/src/api/endpoints/signup/signup.ts (hunk 1 above)
Comment (cond0r) [reply packages/bigcommerce/src/api/endpoints/signup/signup.ts:36]: This displays the actual error and if there are multiple errors. And I checked, and the BigCommerce message was more explicit. I modified this because BigComerce might throw many invalid requests/validations that, at signup, showed it a generic 500 error message for a password that was seven characters but not a numbers & letters combination. It still catches the duplicated email.

File: packages/commerce/src/schemas/customer.ts (hunk 6 above)
Comment (cond0r) [reply packages/commerce/src/schemas/customer.ts:8]: This I took from another PR that reported that error, but yes, will modify it in the endpoint handler and will make there z.object({customer: customerSchema}), because the operation, returns that as customer obj.

File: packages/commerce/src/api/utils/index.ts (hunk 5 above)
Comment (cond0r) [reply packages/commerce/src/api/utils/index.ts:49]: `req.url` comes with full path + query search, which was lacking before.

File: packages/bigcommerce/src/api/operations/login.ts (hunk 3 above)
Comment (cond0r) [reply packages/bigcommerce/src/api/operations/login.ts:67]: Since the res is not passed after dynamic handler PR, I removed that from here too.

File: packages/bigcommerce/src/api/utils/fetch-graphql-api.ts (hunk 4 above)
Comment (cond0r) [reply packages/bigcommerce/src/api/utils/fetch-graphql-api.ts]: This was aligned with the changes already **merged**, so back then you missed this change on the previous PR. I reverted these changes to reflect your comment with all providers to have headers passed in an options obj.
